src/rdkafka_broker.c

```diff
@@ -343,6 +343,15 @@ void rd_kafka_broker_fail (rd_kafka_broker_t *rkb,
 		rkb->rkb_recv_buf = NULL;
 	}
 
+        /* Reset max blocking time back to the default to avoid busy-looping
+         * on reconnect if blocking=0 (#1397).
+         * But honour the lower on-termination blocking time. */
+        if (rd_kafka_terminating(rkb->rkb_rk))
+                rkb->rkb_blocking_max_ms = 1;
+        else
```

Comment:
my understanding is limited, but this looks good to me (resetting back to what it would be on a new connection). I wonder if you need to explicitly set this to 1 in the termination case though, but doesn't hurt.

Reply:
Worst case the thread will block on io-wait for blocking_max_ms (def 1000ms), this slows down termination, so we minimize the blocking time when we know that we are terminating.